### PR TITLE
feat: Added a validator for forbidden shape_distance

### LINKED ISSUE #1885: Flex: forbidden_shape_dist_traveled
### Describe the problem

[Based on this old feed](https://data.trilliumtransit.com/gtfs/mountadams-wa-us/mountadams-wa-us.zip) and [its associated validation report,](https://staging-gtfs-validator-results.mobilitydata.org/6a9c405f-11ab-4824-9b3d-6d77fa39d13f/report.html), there are cases where a `shape_dist_traveled` value is included in `stop_times.txt` for a location or location group, where it doesn't make sense to draw a shape. 

### Describe the new validation rule

**forbidden_shape_dist_traveled**

Logic: If in `stop_times.txt`, there is a row with either `location_id` or `location_group_id` and `shape_dist_traveled` value, return an error.

Description: A GeoJSON location or location group has an associated shape_dist_traveled field in stop_times.txt. shape_dist_traveled values should only be provided for stops. 

Columns: 

tripId
locationId
locationGroupId
csvRowNumber
shapeDistTraveled

### Sample GTFS datasets

https://data.trilliumtransit.com/gtfs/mountadams-wa-us/mountadams-wa-us.zip
http://data.trilliumtransit.com/gtfs/thurston-wa-us/thurston-wa-us--flex-v2.zip

### Severity

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jcpitre**
**Summary:**
Closes #1885 
Added a validator that checks for stopTime entries with location_id or location_group_id values that also have shape_dist_traveled value. If found, add a ForbiddenShapeDistTraveledNotice notice.

**Expected behavior:** 

Used http://data.trilliumtransit.com/gtfs/thurston-wa-us/thurston-wa-us--flex-v2.zip as a test dataset.

Report before the change:
 
image

---

Report after the change:
image

***
Essentially we created 156 forbidden_shape_dist_traveled notices.
https://github.com/MobilityData/gtfs-validator/pull/1895 should take care of removing the 156 decreasing_or_equal_stop_time_distance notices.

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `gradle test` to make sure you didn't break anything
- [ ] Add or update any needed [documentation](https://github.com/MobilityData/gtfs-validator/tree/master/docs) to the repo 
- [x] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [x] Linked all relevant issues
- [x] Include screenshot(s) showing how this pull request works and fixes the issue(s)

### CODE REVIEW

File: main/src/main/java/org/mobilitydata/gtfsvalidator/validator/StopTimeIncreasingDistanceValidator.java [hunk 1]
```diff
@@ -48,27 +48,40 @@ public class StopTimeIncreasingDistanceValidator extends FileValidator {
...
+      // The last good is the last stopTime entity that has a stopId
+      GtfsStopTime lastGood = null;
+      for (var curr : stopTimeList) {
+        if (!curr.hasStopId()) {
+          continue;
+        }
+        if (lastGood == null) {
+          lastGood = curr;
+          continue;
+        }
```
Comment (davidgamez) [inline main/src/main/java/org/mobilitydata/gtfsvalidator/validator/StopTimeIncreasingDistanceValidator.java:60]: I thought only rows without stop IDs would be ignored. In this implementation, we are potentially increasing the travel distance by not considering the intermediate stops without stop IDs. Is this the expected results? cc: @tzujenchanmbd

Comment (tzujenchanmbd) [reply main/src/main/java/org/mobilitydata/gtfsvalidator/validator/StopTimeIncreasingDistanceValidator.java:60]: > I thought only rows without stop IDs would be ignored. I In this implementation, we are potentially increasing the travel distance by not considering the intermediate stops without stop IDs.

I probably don't understand the question here well.

`decreasing_or_equal_stop_time_distance` notice should only compare `shape_dist_traveled` value for rows with `stop_id`.  (**ignore** rows without `stop_id`)

And there should be another notice (this one) to be triggered if `shape_dist_traveled` value exists (no matter what value it is) in the row AND the row without `stop_id`.

Comment (jcpitre) [reply main/src/main/java/org/mobilitydata/gtfsvalidator/validator/StopTimeIncreasingDistanceValidator.java]: Reverted changes to StopTimeIncreasingDistanceValidator as it's covered in https://github.com/MobilityData/gtfs-validator/pull/1895

**Review by davidgamez — APPROVED** [review]
LGTM!

**Pull request merged**